File: UCFPT/211120/solutions/d/quine.java
/*
  quine.java
  Is the Name of This Problem, MCPC 2012 Problem D
  Java solution by Gabriel Foust
*/

import java.io.*;
import java.util.*;

public class quine
{
    public static void main( String[] args ) throws FileNotFoundException
    {
        Scanner scanner= new Scanner( new File( "quine.in" ) );
        
        String line;

        line= scanner.nextLine();
        while (!line.equals( "END" ))
        {
            if (line.length() >= 5 &&
                line.charAt( 0 ) == '"' &&
                line.charAt( line.length()/2 ) == '"' &&
                line.charAt( line.length()/2 + 1 ) == ' ')
            {
                String A = line.substring( 1, line.length()/2 );
                String B = line.substring( line.length()/2 + 2 );

                if (A.equals( B ) && !A.contains( "\"" ))
                {
                    System.out.printf( "Quine(%s)", A );
                    System.out.println();
                }
                else
                    System.out.println( "not a quine" );
            }
            else
                System.out.println( "not a quine" );

            line= scanner.nextLine();
        }
    }
}

File: UCFPT/211120/solutions/d/quine.cpp
/*
  quine.cpp
  Is the Name of This Problem, MCPC 2012 Problem D
  C++ solution by Gabriel Foust
*/

#include <fstream>
#include <iostream>
#include <string>
using namespace std;

ifstream fin( "quine.in" );

int main()
{
  string line;

  while (getline( fin, line ) && line != "END")
  {
    cerr << "ECHO (" << line.size() << "): " << line << endl;
    if (line.length() >= 5 &&
        line[0] == '"' &&
        line[line.length()/2] == '"' &&
        line[line.length()/2 + 1] == ' ')
    {
      string A= line.substr( 1,                   line.length()/2 - 1);
      string B= line.substr( line.length()/2 + 2, line.length()/2 - 1);
      if (A == B && A.find( "\"" ) == string::npos)
        cout << "Quine(" << A << ")" << endl;
      else
        cout << "not a quine" << endl;
    }
    else
      cout << "not a quine" << endl;
  }

  return 0;
}
